src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -1098,73 +1111,102 @@ private AmqpFrameHeaderFW setTransferFrame(
+                {
+                    amqpPropertiesRW.subject(dataEx.properties().subject());
+                }
+                if (dataEx.properties().hasReplyTo())
+                {
+                    amqpPropertiesRW.replyTo(dataEx.properties().replyTo());
+                }
+                if (dataEx.properties().hasCorrelationId())
+                {
+                    amqpPropertiesRW.correlationId(dataEx.properties().correlationId().stringtype());
+                }
+                if (dataEx.properties().hasContentType())
+                {
+                    amqpPropertiesRW.contentType(dataEx.properties().contentType());
+                }
+                if (dataEx.properties().hasContentEncoding())
+                {
+                    amqpPropertiesRW.contentEncoding(dataEx.properties().contentEncoding());
+                }
+                if (dataEx.properties().hasAbsoluteExpiryTime())
+                {
+                    amqpPropertiesRW.absoluteExpiryTime(dataEx.properties().absoluteExpiryTime());
+                }
+                if (dataEx.properties().hasCreationTime())
+                {
+                    amqpPropertiesRW.creationTime(dataEx.properties().creationTime());
+                }
+                if (dataEx.properties().hasGroupId())
+                {
+                    amqpPropertiesRW.groupId(dataEx.properties().groupId());
+                }
+                if (dataEx.properties().hasGroupSequence())
+                {
+                    amqpPropertiesRW.groupSequence(dataEx.properties().groupSequence());
+                }
+                if (dataEx.properties().hasReplyToGroupId())
+                {
+                    amqpPropertiesRW.replyToGroupId(dataEx.properties().replyToGroupId());
+                }
                 properties = amqpPropertiesRW.build();
             }
             return properties;
         }
 
+        private AmqpMapFW<AmqpValueFW, AmqpValueFW> applicationProperties(
+            ArrayFW<AmqpApplicationPropertyFW> applicationProperties,
+            int bufferOffset)
+        {
+            AmqpMapFW<AmqpValueFW, AmqpValueFW> properties = null;
+            if (applicationProperties.fieldCount() > 0)
+            {
+                amqpMapRW.wrap(extraBuffer, bufferOffset, extraBuffer.capacity());
+                applicationProperties.forEach(pair ->
+                {
+                    AmqpStringFW key = amqpStringRW.wrap(valueBuffer, valueOffset.value, valueBuffer.capacity())
+                        .set(pair.key()).build();
+                    valueOffset.value += key.limit();
+                    AmqpStringFW value = amqpValueRW.wrap(valueBuffer, valueOffset.value, valueBuffer.capacity())
+                        .set(pair.value()).build();
+                    valueOffset.value += value.limit();
+                    amqpMapRW.entry(k -> k.setAsAmqpString(key), v -> v.setAsAmqpString(value));
+                });
```

Comment:
Can this be moved to a method and then pass the method reference in place of the lambda?